crates/runtime/src/accelerated_table/refresh_task/changes.rs

```diff
@@ -157,31 +157,21 @@ impl RefreshTask {
                 ChangeOperation::Create | ChangeOperation::Update | ChangeOperation::Read => {
                     let inner_data: RecordBatch = change_batch.data(row);
                     let primary_keys = change_batch.primary_keys(row);
-                    let ctx = SessionContext::new();
-                    let session_state = ctx.state();
 
                     tracing::info!(
                         "Upserting data row for {dataset_name} with {}",
                         Self::get_primary_key_log_fmt(&inner_data, &primary_keys)?
                     );
 
-                    let insert_plan = self
-                        .accelerator
-                        .insert_into(
-                            &session_state,
-                            Arc::new(DataUpdateExecutionPlan::new(DataUpdate {
-                                schema: inner_data.schema(),
-                                data: vec![inner_data],
-                                update_type: UpdateType::Append,
-                            })),
-                            false,
-                        )
-                        .await
-                        .context(crate::accelerated_table::FailedToWriteDataSnafu)?;
-
-                    collect(insert_plan, ctx.task_ctx())
-                        .await
-                        .context(crate::accelerated_table::FailedToWriteDataSnafu)?;
+                    self.write_data_update(
```

Comment:
This does a lot of unnecessary work that will slow down the CDC stream - since we only get changes one row at a time. Things like calculating the size of the update and marking it as ready. Let's just create the streaming plan here and insert it into the table directly.

Reply:
Updated 👍